jerry-core/jcontext/jcontext.h

```diff
@@ -187,6 +190,36 @@ extern jerry_hash_table_t jerry_global_hash_table;
 
 #endif /* !CONFIG_ECMA_LCACHE_DISABLE */
 
+#else /* JERRY_ENABLE_EXTERNAL_CONTEXT */
+
+#define JERRY_CONTEXT(field) (((jerry_context_t *) jerry_port_get_current_instance()->context_p)->field)
+
+#ifndef JERRY_SYSTEM_ALLOCATOR
+
+#define JERRY_HEAP_CONTEXT(field) (((jmem_heap_t *) jerry_port_get_current_instance ()->lcache_p)->field)
+
+#undef JMEM_HEAP_SIZE
+#define JMEM_HEAP_SIZE (jerry_port_get_current_instance ()->heap_size)
+
+#undef JMEM_HEAP_AREA_SIZE
+#define JMEM_HEAP_AREA_SIZE (jerry_port_get_current_instance ()->heap_size - JMEM_ALIGNMENT)
+
```

Comment:
In this way, the users should provide setter function for `heap`, `context`, `lcache` and `instance (for get the size, row-count, row-length)`. Will it be too complicated for users?